jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java

```diff
@@ -195,6 +187,28 @@ protected void doStart() throws Exception {
     }
 
     super.doStart();
+
+    // Look for a datastore service within the context
+    ClassLoader orig = Thread.currentThread().getContextClassLoader();
+    try {
+      ClassLoader loader = getClassLoader();
+      Thread.currentThread().setContextClassLoader(loader);
+      Class<?> factory = loader.loadClass(DatastoreServiceFactory.class.getName());
+      contextDatastoreService = factory.getMethod("getDatastoreService").invoke(null);
+      if (contextDatastoreService != null) {
+        getActiveTransactions =
+            contextDatastoreService.getClass().getMethod("getActiveTransactions");
+        getActiveTransactions.setAccessible(true);
```

Comment:
When I wrote the code I didn't have it, but it had permission problems that the setAccessible fixed.   I didn't research exactly why.  I can try again without and look in more detail what permission is failing and why.